subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/observability/BuildScanFacade.kt

```diff
@@ -0,0 +1,169 @@
+
+    fun instantExecutionStoreAction() =
+        value("instant-execution:action", "store")
+
+    fun instantExecutionLoadAction() =
+        value("instant-execution:action", "load")
+
+    fun instantExecutionStateSize(size: Long) =
+        value("instant-execution:state:size", size)
+
+    fun instantExecutionFingerprintSize(size: Long) =
+        value("instant-execution:fingerprint:size", size)
+}
+
+
+internal
+class DefaultBuildScanFacade(
+    private val host: InstantExecutionHost,
+    private val buildScanPluginApplied: BuildScanPluginApplied
+) : BuildScanFacade {
+
+    private
+    val gradleEnterprise: Any by unsafeLazy {
+        host.currentBuild.gradle.settings.extensionNamed("gradleEnterprise")
+    }
+
+    private
+    val buildScan: Any by unsafeLazy {
+        gradleEnterprise.withGroovyBuilder { getProperty("buildScan") }
+    }
+
+    override var enterpriseServer: String?
+        by stringGroovyProperty("server", lenient = true) { gradleEnterprise }
+
+    override var termsOfServiceUrl: String?
+        by stringGroovyProperty { buildScan }
+
+    override var termsOfServiceAgree: String?
+        by stringGroovyProperty { buildScan }
+
+    override var server: String?
+        by stringGroovyProperty { buildScan }
+
+    override var isAllowUntrustedServer: Boolean
+        by booleanGroovyProperty("allowUntrustedServer") { buildScan }
+
+    override var isCaptureTaskInputFiles: Boolean
+        by booleanGroovyProperty("captureTaskInputFiles") { buildScan }
+
+    override fun tag(tag: String): Unit = buildScan {
+        "tag"(tag)
+    }
+
+    override fun value(key: String, value: Any?): Unit = buildScan {
+        "value"(key, value.toString())
+    }
+
+    private
+    val isBuildScanEnabled: Boolean
+        get() = buildScanPluginApplied.isBuildScanPluginApplied
+
+    private
+    fun Any.extensionNamed(name: String): Any =
+        (this as ExtensionAware).extensions.getByName(name)
+
+    private
+    fun stringGroovyProperty(name: String? = null, lenient: Boolean = false, owner: () -> Any) =
+        GroovyPropertyDelegate(name, lenient, owner) { it as? String }
+
+    private
+    fun booleanGroovyProperty(name: String? = null, lenient: Boolean = false, owner: () -> Any) =
+        GroovyPropertyDelegate(name, lenient, owner) { it as? Boolean ?: false }
+
+    private
+    fun buildScan(dynamicAction: GroovyBuilderScope.() -> Unit) =
+        if (isBuildScanEnabled) buildScan.withGroovyBuilder(dynamicAction)
+        else Unit
+
+    /**
+     * GroovyBuilder based Kotlin delegated property.
+     *
+     * @param name Overridden property name, or null to use the Kotlin property name
+     * @param lenient If `true`, errors accessing dynamic properties are ignored
+     * @param owner Provides the owner of the dynamic properties
+     * @param transform Transforms values going in and out
+     */
+    private
+    inner class GroovyPropertyDelegate<T : Any?>(
```

Comment:
🙈

Reply:
Yes 🙈 Hopefully this should be a temporary measure.